README.md

```diff
@@ -256,6 +256,28 @@ message like this:
 `rpc error: code = Unauthenticated desc = Unauthorized: HTTP status code 401`.
 To generate a new token, rerun `source auth/CLOUDRUN.sh`.
 
+## Running the Registry API server on GKE
+
+The [Makefile](Makefile) contains targets that build a Docker image
+(`make build`) and that deploy it to GKE (`make deploy-gke`).
+
+Requirements:
+
+- Ensure you have [gcloud](https://cloud.google.com/sdk/gcloud) and
+  [kubectl](https://cloud.google.com/kubernetes-engine/docs/quickstart)
+  installed.
+
+- If not already done, `gcloud auth login` gets user credentials for subsequent
+  `gcloud` operations and `gcloud config set project PROJECT_ID` can be used to
+  set your project ID to the one where you plan to host your servce.
+
+- The Makefile gets your project ID from the `REGISTRY_PROJECT_IDENTIFIER`
+  environment variable. This can be set automatically by running
+  `source auth/GKE.sh`.
```

Comment:
Not sure how this can be best addressed but to do `make build` you need `REGISTRY_PROJECT_IDENTIFIER` which can be set by running `source auth/GKE.sh`. But in `GKE.sh`, we are trying to get `ingress _ip` and `service port` which requires the deployment to happen first and to deploy we need to build first.

Kind of a circular dependency. When I tried this in my local, instead of running `source auth/GKE.sh` to set the `REGISTRY_PROJECT_IDENTIFIER`, I just ran `export REGISTRY_PROJECT_IDENTIFIER=$(gcloud config list --format 'value(core.project)')`.

Reply:
Yes I had the same question while setting up for CLOUDRUN and running auth/CLOUDRUN.sh while I didn't have any instance created. I feel this worth a separate CL to put `REGISTRY_PROJECT_IDENTIFIER=$(gcloud config list --format 'value(core.project)')` into the `makefile` itself, so the `makefile` is separated from `auth/.*.sh`. 

To temporarily solve this, I modified `auth/GKE.sh` so it returns while there's no GKE cluster exists or no `ingress_ip` available. But we do need a refactor on this

Reply:
Thanks for addressing this. We should create a ticket to track the follow up required though.

Reply:
Sometimes I've worked around this by running `auth/LOCAL.sh`, but realize now - that was just a hack.

Reply:
I created a new issue: https://github.com/apigee/registry/issues/35. I'll address separately